node/src/components/storage.rs

```diff
@@ -922,6 +927,11 @@ impl Storage {
             } => responder
                 .respond(self.store_finalized_approvals(deploy_hash, finalized_approvals)?)
                 .ignore(),
+            StorageRequest::GetSwitchBlockHeightAtEraId { era_id, responder } => responder
+                .respond(
+                    self.get_switch_block_height_by_era_id(&mut self.env.begin_ro_txn()?, era_id)?,
+                )
+                .ignore(),
```

Comment:
We can avoid the `RefCell` if we just inline the code something like this:
```suggestion
            StorageRequest::GetSwitchBlockHeightAtEraId { era_id, responder } => {
                let mut txn = self.env.begin_ro_txn()?;
                if let Some(block_height) = self.switch_block_height_by_era_id_cache.get(&era_id) {
                    return Ok(responder.respond(Some(*block_height)).ignore());
                }
                let block_header = match self.get_switch_block_header_by_era_id(&mut txn, era_id)? {
                    Some(block_header) => block_header,
                    None => return Ok(responder.respond(None).ignore()),
                };
                self.switch_block_height_by_era_id_cache
                    .insert(era_id, block_header.height());
                responder.respond(Some(block_header.height())).ignore()
            }
```

Reply:
Nice one, added. I didn't see how I can remove RefCell at first